### PR TITLE
Feature/example values

### PR DESCRIPTION
**Pull request by fehguy**
Adds example values for query, header, form, and path params.  The goal is to differentiate between the `default` value, which is intended to be "what the server will do by default if no value is passed".  The example construct is not supported in the 2.0 specification so it is being exposed using `x-example` for scalar parameters and `x-examples` for body parameters.

For example:

```java
@ApiOperation(value = "test")
@GET
@Path("external/info")
public void queryExample(@ApiParam(value = "test",
        example = "a,b,c") @QueryParam("tenantId") ArrayList<String> tenantId) {
    return;
}
```

This is represented as:
```json
{
  "name" : "tenantId",
  "in" : "query",
  "description" : "test",
  "required" : false,
  "type" : "a,b,c",
  "items" : {
    "type" : "string"
  },
  "x-example" : "a,b,c"
}
```

will allow the string array to have an example value of `a, b, c`

Likewise for a post body:

```java
@ApiOperation(value = "test")
@POST
@Path("external/info")
public void postTest(@ApiParam(value = "test",
        examples = @Example(value = {
                @ExampleProperty(mediaType="application/json", value="[\"a\",\"b\"]")
        })) ArrayList<String> tenantId) {
    return;
}
```

This is represented as:
```json
{
  "in" : "body",
  "name" : "body",
  "description" : "test",
  "required" : false,
  "schema" : {
    "type" : "array",
    "items" : {
      "type" : "string"
    }
  },
  "x-examples" : {
    "application/json" : "[\"a\",\"b\"]"
  }
}
```

### CODE REVIEW

File: modules/swagger-annotations/src/main/java/io/swagger/annotations/Example.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+@Target(ElementType.ANNOTATION_TYPE)
+@Retention(RetentionPolicy.RUNTIME)
+public @interface Example {
+
+    /**
+     * An option name for these examples.
+     *
+     * @return an option name for these examples - will be prefixed with "x-"
+     */
+    String name() default "";
```
Comment (webron) [inline modules/swagger-annotations/src/main/java/io/swagger/annotations/Example.java]: This is not used by the code.

File: modules/swagger-core/src/main/java/io/swagger/util/ParameterProcessor.java [hunk 2]
```diff
@@ -341,6 +382,13 @@ public ApiParam getAnnotation() {
         public boolean isHidden() {
             return apiParam.hidden();
         }
+
+        @Override
+        public String getExample() {
+            return apiParam.example();
+        };
+
+        public Example getExamples() { return apiParam.examples();};
```
Comment (webron) [inline modules/swagger-core/src/main/java/io/swagger/util/ParameterProcessor.java]: Shouldn't be a single line.

File: modules/swagger-core/src/main/java/io/swagger/util/ParameterProcessor.java [hunk 3]
```diff
@@ -408,5 +456,12 @@ public ApiImplicitParam getAnnotation() {
         public boolean isHidden() {
             return false;
         }
+
+        @Override
+        public String getExample() {
+            return apiParam.example();
+        };
+
+        public Example getExamples() { return apiParam.examples();};
```
Comment (webron) [inline modules/swagger-core/src/main/java/io/swagger/util/ParameterProcessor.java]: Shouldn't be a single line.

**Comment by fehguy** [comment]
I've addressed the two comments from before.

**Comment by fehguy** [comment]
Need to make sure the example value has the correct type, currently always `string`.

**Pull request merged**